database/multistmt/parse.go

```diff
@@ -58,21 +44,71 @@ func Parse(reader io.Reader, delimiter []byte, maxMigrationSize int, replacement
 	// completed statements, contents of accum will be dumped in here
 	stmts := make([][]byte, 0, 1000)
 
+	tmp := make([]byte, 0, 10)
+	a := 0
 	for err == nil {
+		// this is non-performant but very clear that the buf variable is wiped out
+		//completely between loop iterations
+		buf = make([]byte, ParseBufSize)
 		n, err := reader.Read(buf)
+		trace("tmp(2): '%s', buf: %s, discard: %v\n", tmp, buf, discard)
+		// tmp is the carry-over buffer,
+		//if the previous loop iteration had two few characters to make comparisions,
+		//tmp will have the characters at the point the loop iteration was abandoned(
+		//break'd out of)
+		if len(tmp) > 0 {
+			trace("copying '%s' to buf\n", tmp)
```

Comment:
```suggestion
			trace("prepending '%s' to buf\n", tmp)
```

to be clear what exactly we are doing